UOP1_Project/Assets/Scripts/Characters/StateMachine/Actions/ActiveInteractionResponse_OnEnterSO.cs

```diff
@@ -0,0 +1,32 @@
+﻿using UnityEngine;
+using UOP1.StateMachine;
+using UOP1.StateMachine.ScriptableObjects;
+
+[CreateAssetMenu(fileName = "ActiveInteractionResponse_OnEnter", menuName = "State Machines/Actions/Active Interaction Response On Enter")]
+public class ActiveInteractionResponse_OnEnterSO : StateActionSO
+{
+	public VoidEventChannelSO interactResponse;
+	protected override StateAction CreateAction() => new ActiveInteractionResponse_OnEnter();
+}
+
+public class ActiveInteractionResponse_OnEnter : StateAction
+{
+	protected new ActiveInteractionResponse_OnEnterSO OriginSO => (ActiveInteractionResponse_OnEnterSO)base.OriginSO;
+	
+	public override void Awake(StateMachine stateMachine)
+	{
+	}
+	
+	public override void OnUpdate()
+	{
+		OriginSO.interactResponse.RaiseEvent();
```

Comment:
This should probably be in `OnStateEnter`

Reply:
Yes, this Should be o_O', Thank you